File: generators/app/templates/middle-tier/src/main/groovy/spring-mvc/service/UserServiceImpl.java
package <%= packageName %>.service;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import <%= packageName %>.domain.User;
import <%= packageName %>.repository.UserRepository;

@Service
public class UserServiceImpl implements UserService {

    @Autowired
    private UserRepository userRepository;

    @Override
    public Optional<User> user(String username) {
        return userRepository.readUser(username);
    }

    @Override
    public List<User> users() {
        return userRepository.readUsers();
    }

}

File: generators/app/templates/middle-tier/src/main/kotlin/spring-mvc/service/UserServiceImpl.kt
package <%= packageName %>.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

import <%= packageName %>.domain.User
import <%= packageName %>.repository.UserRepository

@Service
class UserServiceImpl : UserService {

    @Autowired
    private val userRepository: UserRepository? = null

    override fun users(): Collection<User> {
        return userRepository!!.readUsers()
    }

    override fun user(username: String): User? {
        return userRepository!!.readUser(username)
    }

}
